arcade/arcade/worker/mcp/sse.py

```diff
@@ -0,0 +1,447 @@
+
+        # Make sure the message path ends with a slash for proper mounting
+        message_mount_path = self.message_path
+        if not message_mount_path.endswith("/"):
+            message_mount_path = f"{message_mount_path}/"
+
+        # Make sure the message path doesn't start with a slash for proper URL joining
+        if message_mount_path.startswith("/"):
+            message_mount_path = message_mount_path[1:]
+
+        # Create Starlette app for SSE
+        starlette_app = Starlette(
+            debug=True,
+            routes=[
+                Route(self.sse_path, endpoint=self._handle_sse),
+                Mount(f"/{message_mount_path}", app=self.sse.handle_post_message),
+            ],
+        )
+
+        # Mount Starlette app to FastAPI app
+        self.fastapi_app.mount("/mcp", starlette_app)
+        self.initialized = True
+        logger.info(f"MCP SSE server initialized at /mcp{self.sse_path}")
+        logger.info(f"MCP message endpoint available at /mcp/{message_mount_path}")
+        logger.info(f"Configured for up to {self.max_connections} simultaneous connections")
+
+    def _get_connection_id(self) -> str:
+        """Generate a unique connection ID and track it."""
+        with self.connection_lock:
+            if len(self.active_connections) >= self.max_connections:
+                # Too many connections
+                logger.warning(
+                    f"Maximum connections ({self.max_connections}) reached, rejecting new connection"
+                )
+                return ""
+
+            self.connection_count += 1
+            conn_id = f"sse_{self.connection_count}"
+            self.active_connections.add(conn_id)
+            logger.debug(
+                f"New connection {conn_id} established. Active connections: {len(self.active_connections)}"
+            )
+            return conn_id
+
+    def _release_connection_id(self, conn_id: str) -> None:
+        """Remove a connection ID from tracking."""
+        with self.connection_lock:
+            if conn_id in self.active_connections:
+                self.active_connections.remove(conn_id)
+                logger.debug(
+                    f"Connection {conn_id} released. Active connections: {len(self.active_connections)}"
+                )
+
+    async def _list_tools(self) -> list[MCPTool]:
+        """Handler for the MCP list_tools operation."""
+        tools = []
+        for definition in self.worker.get_catalog():
+            try:
+                # Parse the fully qualified name (toolkit.name) into toolkit and name parts
+                parts = definition.fully_qualified_name.split(".")
+                if len(parts) == 2:
+                    toolkit_name, tool_name = parts
+                    # Create a new FullyQualifiedName object with the parsed parts
+                    fqn = FullyQualifiedName(tool_name, toolkit_name)
+                    tool = self.worker.catalog[fqn]
```

Comment:
Is this part necessary, since you are already looping over the definitions from `get_catalog()`?